config.go

```diff
@@ -106,6 +108,27 @@ func (c *Config) Discover() error {
 		}
 	}
 
+	// Finally, we'll fill in any internal plugins that haven't already been
+	// discovered on disk.
```

Comment:
Yeah this is a good point. There was a bit of pain around this in Packer so we added a note to the [download page](https://www.packer.io/downloads.html) and also called this out in the changelog.

Ideally we could detect that the plugin is old but there is currently no plugin versioning so this isn't something we can check.

What I could do is add versioning going forward and we can complain in the UI if the plugin doesn't have a version specified. This might involve a change to hashicorp/go-plugin.